tests/unit_tests/test_target.py

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env python3
+##############################################################################
+# (c) Crown copyright Met Office. All rights reserved.
+# For further details please refer to the file COPYRIGHT
+# which you should have received as part of this distribution
+##############################################################################
+
+"""
+Unit tests for fab recipe classes.
+"""
+
+import argparse
+import pytest
+
+from fab.target.base import FabTargetBase
+from fab.target.zero import FabZeroConfig
+
+
+def test_valid_subclass():
+    """Create a valid subclass from FabTargetBase."""
+
+    class FabTarget(FabTargetBase):
+        """Trivial test class."""
+
+        project_name = "test-class"
+
+        def run(self, _):
+            """Dummy run method."""
+
+    assert issubclass(FabTarget, FabTargetBase)
+
+    target = FabTarget()
+    assert isinstance(target, FabTargetBase)
+
+
+def test_invalid_subclasses():
+    """Create invalid subclasses from FabTargetBase."""
+
+    class FabInvalidMethod(FabTargetBase):
+        """Missing abstract methods."""
+
+        project_name = "test-class"
+
+    with pytest.raises(TypeError) as exc:
+        _ = FabInvalidMethod()
+    assert "Can't instantiate abstract class" in str(exc)
+
+    class FabInvalidProject(FabTargetBase):
+        """Missing project_name property."""
+
+        def run(self, _):
+            """Dummy run method."""
+
+    with pytest.raises(TypeError) as exc:
+        _ = FabInvalidMethod()
+    assert "Can't instantiate abstract class" in str(exc)
+
+
+def test_add_arguments():
+    """Test add_arguments static method."""
+
+    class FabTargetArgs(FabTargetBase):
+        """Test class with add_arguments."""
+
+        @staticmethod
+        def add_arguments(p):
+            """Add a positional argument."""
+            p.add_argument("test", type=str)
+
+    parser = argparse.ArgumentParser()
```

Comment:
It feel very strange that the application has to create and provide its own `ArgumentParser`. Actually, this whole test is a bit odd: you basically call a static function in `FabTargetArgs`. Does deriving from `FabTargetBase` actually make any difference, or is any code of `FabTargetBase` actually used in this test? To me it looks like it is only testing `argparse`'s `add_argument` method, and nothing else. Am I missing something?